packages/core/src/libraries/jwt-customizer.ts

```diff
@@ -49,29 +49,23 @@ export const createJwtCustomizerLibrary = (queries: Queries, userLibrary: UserLi
           };
         })
       ),
-      // No need to deal with the type here, the type will be enforced by the guard when return the result.
-      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-      organizations: Object.fromEntries(
-        await Promise.all(
-          organizationsWithRoles.map(async ({ organizationRoles, ...organization }) => [
-            organization.id,
-            {
-              roles: await Promise.all(
-                organizationRoles.map(async ({ id, name }) => {
-                  const [_, fullOrganizationScopes] = await relations.rolesScopes.getEntities(
-                    OrganizationScopes,
-                    { organizationRoleId: id }
-                  );
-                  return {
-                    id,
-                    name,
-                    scopes: fullOrganizationScopes.map(pickState('id', 'name', 'description')),
-                  };
-                })
-              ),
-            },
-          ])
-        )
+      organizations: await Promise.all(
+        organizationsWithRoles.map(async ({ organizationRoles, ...organization }) => ({
+          id: organization.id,
+          roles: await Promise.all(
```

Comment:
this looks slow. if a user has 100 orgs with 3 roles per org it will be 300 db queries at the same time. i suggest either:

- don't append scopes here for now
- query all roles first, directly map scopes accordingly to user's roles.  then only one promise is needed

Reply:
Good call.
Updated by querying roles and mapping them to the org. Updated normal user role/scope as well.